src/mlpack/methods/grad_boosting/grad_boosting.hpp

```diff
@@ -0,0 +1,217 @@
+   */
+  void Train(const MatType& data,
+             const data::DatasetInfo& datasetInfo,
+             const arma::Row<size_t>& labels,
+             const size_t numClasses,
+             const size_t numWeakLearners);
+
+  /**
+   * Train Gradient Boosting on the given dataset, using the given parameters.
+   * The last parameters are the hyperparameters to use for the weak learners;
+   * these are all the arguments to `WeakLearnerType::Train()` after `numClasses`
+   * and `weights`.
+   *
+   * Default values are not used for `numWeakLearners`; instead, it is used to specify
+   * the number of weak learners (models) to train during gradient boosting.
+   *
+   * @param data Dataset to train on.
+   * @param labels Labels for each point in the dataset.
+   * @param numClasses The number of classes in the dataset.
+   * @param numWeakLearners Number of boosting rounds.
+   * @param minimumLeafSize Minimum number of points in each leaf node.
+   * @param minimumGainSplit Minimum gain for the node to split.
+   * @param maximumDepth Maximum depth for the tree.
+   */
+  void Train(const MatType& data,
+             const data::DatasetInfo& datasetInfo,
+             const arma::Row<size_t>& labels,
+             const size_t numClasses,
+             const size_t numWeakLearners,
+             const size_t minimumLeafSize,
+             const double minimumGainSplit,
+             const size_t maximumDepth);
+
+  /**
+   * Classify the given test point.
+   *
+   * @param point Test point.
+   */
+  template<typename VecType>
+  size_t Classify(const VecType& point);
+
+  /**
+   * Classify the given test point.
+   *
+   * @param point Test point.
+   * @param prediction Will be filled with the predicted class of `point`.
+   */
+  template<typename VecType>
+  void Classify(const VecType& point,
+                size_t& prediction);
+
+  /**
+   * Classify the given test points.
+   *
+   * @param test Testing data.
+   * @param predictedLabels Vector in which the predicted labels of the test
+   *      set will be stored.
+   */
+  void Classify(const MatType& test,
+                arma::Row<size_t>& predictedLabels);
+
+  /**
+   * Serialize the GradBoosting model.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const uint32_t version);
+
+ private:
+  /**
+   * Internal utility training function.  
+   */
+  void TrainInternal(const MatType& data,
+                     const data::DatasetInfo& datasetInfo,
+                     const arma::Row<size_t>& labels,
+                     const size_t numWeakLearners = 5,
+                     const size_t minimumLeafSize = 10,
+                     const double minimumGainSplit = 1e-7,
+                     const size_t maximumDepth = 2);
+
+  //! The number of classes in the model.
+  size_t numClasses;
+  //! The number of weak learners in the model.
+  size_t numWeakLearners;
+  //! Adjustments vector.
+  arma::vec adjustments;
```

Comment:
Should we make these weights accessible to the user via an `Adjustments()` function or similar?  What do you think?

Reply:
The adjustment vector was an old version thing - I was trying something, and forgot to delete it 😅